core/conversion/converters/impl/conv_deconv.cpp

```diff
@@ -9,74 +9,147 @@ namespace conversion {
+                nvinfer1::IDeconvolutionLayer* deconv;
+                if (args[2].IValue()->isTensor()) {
+                  Weights b(ctx, args[2].IValue()->toTensor());
+                  deconv = ctx->net->addDeconvolutionNd(*in, w.num_input_maps, w.kernel_shape, w.data, b.data);
+                } else {
+                  deconv = ctx->net->addDeconvolutionNd(*in, w.num_input_maps, w.kernel_shape, w.data, {});
+                }
+
+                TRTORCH_CHECK(deconv, "Unable to create deconvolution layer from node: " << *n);
+
+                deconv->setStrideNd(stride);
+                deconv->setPaddingNd(padding);
+#if NV_TENSORRT_MAJOR > 7 || (NV_TENSORRT_MAJOR == 7 && NV_TENSORRT_MINOR == 1)
+                deconv->setDilationNd(dilation);
+                deconv->setNbGroups(groups);
+#endif
+                new_layer = deconv;
+              } else {
+                nvinfer1::IConvolutionLayer* conv;
+                if (args[2].IValue()->isTensor()) {
+                  Weights b(ctx, args[2].unwrapToTensor());
+                  conv = ctx->net->addConvolutionNd(*in, w.num_output_maps, w.kernel_shape, w.data, b.data);
+                } else {
+                  Weights b(ctx, torch::zeros(args[1].unwrapToTensor().sizes()[0]));
+                  conv = ctx->net->addConvolutionNd(*in, w.num_output_maps, w.kernel_shape, w.data, b.data);
+                }
+
+                TRTORCH_CHECK(conv, "Unable to create convolution layer from node: " << *n);
+
+                conv->setStrideNd(stride);
+                conv->setPaddingMode(nvinfer1::PaddingMode::kCAFFE_ROUND_DOWN);
+                conv->setPaddingNd(padding);
+                conv->setPostPadding(out_padding);
+                conv->setDilationNd(dilation);
+                conv->setNbGroups(groups);
+                new_layer = conv;
+              }
+              new_layer->setName(util::node_info(n).c_str());
+
+              auto out = ctx->AssociateValueAndTensor(n->outputs()[0], new_layer->getOutput(0));
+
+              LOG_DEBUG("Output tensor shape: " << out->getDimensions());
+
+              return true;
+            }})
+        .pattern({
+            R"SIG(aten::_convolution.deprecated(Tensor input, Tensor weight,
+                                     Tensor? bias, int[] stride, int[] padding,
+                                     int[] dilation, bool transposed,
+                                     int[] output_padding, int groups, bool benchmark,
+                                     bool deterministic, bool cudnn_enabled) -> (Tensor))SIG",
+            [](ConversionCtx* ctx, const torch::jit::Node* n, args& args) -> bool {
+              // This pattern is only matched for traced JIT models which do not have allow_tf32 bool
+              // in the function signature. The TRT conversion code is exactly same as the above call.
+              // Is there a better way to match two different patterns for a single TRT conversion call?
```

Comment:
We can use a utility function to unify the implementations

Reply:
Done